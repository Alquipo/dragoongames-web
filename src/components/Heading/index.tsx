import * as S from './styles'

const Heading = () => (
  <S.Wrapper>
    <h1>Heading</h1>
  </S.Wrapper>
)

export default Heading
