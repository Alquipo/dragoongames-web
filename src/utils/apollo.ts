import { ApolloClient, HttpLink, NormalizedCacheObject } from '@apollo/client'
import { useMemo } from 'react'
import apolloCache from './apolloCache'

let apolloClient: ApolloClient<NormalizedCacheObject>

function createApolloClient() {
  return new ApolloClient({
    ssrMode: typeof window === 'undefined',
    link: new HttpLink({ uri: 'http://localhost:1337/graphql' }),
    cache: apolloCache
  })
}

export function initializeApollo(initialState = {}) {
  //serve para verificar se já existe uma instância, para nao criar outra
  const apolloClientGlobal = apolloClient ?? createApolloClient()

  //recuperando os dados de cache
  if (initialState) {
    apolloClientGlobal.cache.restore(initialState)
  }

  //sempre inicializando no SSR com cache limpo
  if (typeof window === 'undefined') return apolloClientGlobal
  apolloClient = apolloClient ?? apolloClientGlobal

  return apolloClient
}

export function useApollo(initialState = {}) {
  const store = useMemo(() => initializeApollo(initialState), [initialState])

  return store
}
