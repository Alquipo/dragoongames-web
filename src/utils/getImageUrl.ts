export const getImageUrl = (url: string | undefined) => {
  if (process.env.NEXT_PUBLIC_IMAGE_HOST) {
    return `${process.env.NEXT_PUBLIC_IMAGE_HOST}${url}`
  }

  if (url) {
    return url
  }

  return null
}
