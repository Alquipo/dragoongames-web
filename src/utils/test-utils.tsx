import { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import {
  CartContext,
  CartContextData,
  CartContextDefaultValues
} from 'hooks/use-cart'

import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'

type CustomRenderProps = {
  cartProviderProps?: CartContextData
} & Omit<RenderOptions, 'queries'>

const customRender = (
  ui: ReactElement,
  {
    cartProviderProps = CartContextDefaultValues,
    ...renderOptions
  }: CustomRenderProps = {}
) =>
  render(
    <ThemeProvider theme={theme}>
      <CartContext.Provider value={cartProviderProps}>
        {ui}
      </CartContext.Provider>
    </ThemeProvider>,
    renderOptions
  )

export * from '@testing-library/react'
export { customRender as render }
