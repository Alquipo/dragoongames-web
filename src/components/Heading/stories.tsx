import { Story, Meta } from '@storybook/react/types-6-0'
import Heading from '.'

export default {
  title: 'Heading',
  component: Heading
} as Meta

export const Default: Story = () => <Heading />
