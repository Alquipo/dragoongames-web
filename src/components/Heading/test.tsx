import { render, screen } from 'utils/test-utils'

import Heading from '.'

describe('<Heading />', () => {
  it('should render a white heading by default', () => {
    render(<Heading>Dragoon Games</Heading>)
    expect(screen.getByRole('heading', { name: /Dragoon games/i })).toHaveStyle(
      {
        color: '#FAFAFA'
      }
    )
  })

  it('should render a black heading when color is passed', () => {
    render(<Heading color="black">Dragoon Games</Heading>)
    expect(screen.getByRole('heading', { name: /Dragoon games/i })).toHaveStyle(
      {
        color: '#030517'
      }
    )
  })

  it('should render a heading with a line to the left side', () => {
    render(<Heading lineLeft>Dragoon Games</Heading>)
    expect(screen.getByRole('heading', { name: /Dragoon games/i })).toHaveStyle(
      {
        'border-left': '0.7rem solid #EB46CE'
      }
    )
  })

  it('should render a heading with a line at the bottom', () => {
    render(<Heading lineBottom>Dragoon Games</Heading>)
    expect(
      screen.getByRole('heading', { name: /Dragoon games/i })
    ).toHaveStyleRule('border-bottom', '0.5rem solid #EB46CE', {
      modifier: '::after'
    })
  })

  it('should render a heading with a small size', () => {
    render(<Heading size="small">Dragoon Games</Heading>)
    expect(screen.getByRole('heading', { name: /Dragoon games/i })).toHaveStyle(
      {
        'font-size': '1.6rem'
      }
    )

    expect(
      screen.getByRole('heading', { name: /Dragoon games/i })
    ).toHaveStyleRule('width', '3rem', {
      modifier: '::after'
    })
  })

  it('should render a heading with a huge size', () => {
    render(<Heading size="huge">Dragoon Games</Heading>)

    expect(screen.getByRole('heading', { name: /Dragoon games/i })).toHaveStyle(
      {
        'font-size': '5.2rem'
      }
    )
  })

  it('should render a Heading with a primary line color', () => {
    render(
      <Heading lineColor="primary" lineLeft lineBottom>
        Lorem Ipsum
      </Heading>
    )

    const heading = screen.getByRole('heading', { name: /lorem ipsum/i })
    expect(heading).toHaveStyle({ 'border-left': '0.7rem solid #EB46CE' })
    expect(heading).toHaveStyleRule('border-bottom', '0.5rem solid #EB46CE', {
      modifier: '::after'
    })
  })

  it('should render a Heading with a secondary line color', () => {
    render(
      <Heading lineColor="secondary" lineLeft lineBottom>
        Lorem Ipsum
      </Heading>
    )

    const heading = screen.getByRole('heading', { name: /lorem ipsum/i })
    expect(heading).toHaveStyle({ 'border-left': '0.7rem solid #1f60ca' })
    expect(heading).toHaveStyleRule('border-bottom', '0.5rem solid #1f60ca', {
      modifier: '::after'
    })
  })
})
