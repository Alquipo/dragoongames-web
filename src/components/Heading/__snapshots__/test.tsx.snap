// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    Heading
  </h1>
</main>
`;
